### PR TITLE
Update site branding information.

### PR DESCRIPTION
**Pull request by zpChris**
### List of Changes:
- Rename the `groupId` from `rocketden` to `codejoust`, and make this change across the repository. (Update the `README.md` slightly as well.)
- Add Open Graph and regular sharing / social link information.
  - I used [this link](https://lh5.googleusercontent.com/XSUz2fZ6Gg4sIl9g-_3JZDJHrClonmJoRHvEpW8Aa_2tKhrYtpZ-JLxEMRkQm--8plOwxD_y1YbkV9taqgdW=w2559-h1328-rw) to directly share the image, since I ran into issues when trying to reference the "public" folder image, as I believe it has to be referenced from server-side. See this [SO post](https://stackoverflow.com/questions/43072222/how-to-include-meta-tags-dynamically-in-react-js); let me know your thoughts @alankbi - I added this solution simply because I wanted to have it at least working before I requested you.
- Change the GitHub and Trello links (see the `ContactUs.tsx` file).
<details><summary>Make general changes to clean up old errors in files (click here to expand the list)</summary>
<ul>
 <li>RoomService.java: Remove unnecessary assignment and break up the cognitive complexity of updateRoomSettingsSelectedProblems.
  Unnecessary Assignment Warning</li>
 <li>ProblemService.java: Update the Exception thrown and change unnecessary replaceAll to replace.
Unnecessary replaceAll Call</li>
 <li>pom.xml: Update out-of-date dependency versions. (Note: There is one remaining warning with log4j, but there is no recommended solution.

  Out of Date Log4j</li>
</ul>
</details>

### Notes:
- We may want to make changes to the `README.md` file that indicate that the other repository `tester` is private, and you thus cannot fully run this repository locally.
- The frontpage logo in the header does not match the usual scale of logo-to-text that we have on the Figma - should the logo be much bigger, or does that look ok?
- This is obviously a huge number of changes for a small amount of work, since I just changed `rocketden` to `codejoust` - maybe at some point in the future @alankbi or @jasonz6688 can take a task like this (maybe for the `tester` repository?), so I don't benefit from these easy line additions. Not sure how important that is, but though I'd mention it.
- I tested the meta tags by temporarily deploying the site and using [Meta Tags](https://metatags.io), and you can test locally with the [Open Graph Preview Chrome Extension](https://chrome.google.com/webstore/detail/open-graph-preview/ehaigphokkgebnmdiicabhjhddkaekgh).

### Screencast and Images:

**Frontpage with Logo and Favicon**
Frontpage with Logo and Favicon

**Meta Tags for Social**
_I'm not certain if these are fully effective on all platforms (when I tested on Slack I still had the "React" favicon for some reason, and when testing on Discord no image showed), but this is at least better than the current setup. I tested the below by temporarily deploying the site and using [Meta Tags](https://metatags.io), and you can test locally with the [Open Graph Preview Chrome Extension](https://chrome.google.com/webstore/detail/open-graph-preview/ehaigphokkgebnmdiicabhjhddkaekgh).
Meta Tags for Social

### CODE REVIEW

**Comment by zpChris** [comment]
[Update site with logo, description, meta text, license, and more](https://trello.com/c/g8skRHTi/333-update-site-with-logo-description-meta-text-license-and-more)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+### Contributing to CodeJoust
+You are free to contribute to the `main` branch of CodeJoust. If you notice any issues on our website, feel free to add an issue or submit your own pull request. If you have any questions, contact us at [support@codejoust.co](support@codejoust.courl).
```
Comment (alankbi) [inline CONTRIBUTING.md:2]:
```suggestion
You are free to contribute to the `main` branch of CodeJoust. If you notice any issues on our website, feel free to add an issue or submit your own pull request. If you have any questions, contact us at [support@codejoust.co](mailto:support@codejoust.co).
```

**Review by alankbi — APPROVED** [review]
Looks good! However, before merging, I think it might be good to go over the license choice at standup or on Discord. Looking at a [quick summary](https://gist.github.com/kn9ts/cbe95340d29fc1aaeaa5dd5c059d2e60) I think I'd agree on the current choice but we should make sure that we fully understand it before committing (e.g. [this clause](https://softwareengineering.stackexchange.com/questions/131264/what-constitutes-distributing-for-lgpl-v3) which seems to work out for us but prob good to still double-check)

File: CONTRIBUTING.md (hunk 1 above)
Comment (zpChris) [reply CONTRIBUTING.md:2]: After our discussion, I'll remove the license so there is simply [no license](https://choosealicense.com/no-permission/) so our work is under exclusive copyright.

**Pull request merged**